designer/report-designer/src/main/java/org/pentaho/reporting/designer/core/actions/elements/MorphAction.java

```diff
@@ -106,6 +112,10 @@ public void actionPerformed( final ActionEvent e ) {
       if ( ObjectUtilities.equal( visualElement.getElementTypeName(), targetElementType.getMetaData().getName() ) ) {
         continue;
       }
+      if ( COMPLEX_ELEMENT_SET.contains( targetElementType.getMetaData().getName() )
```

Comment:
Shouldn't this logic be in the construction of the Morph menu ? It looks to me like they will still be able to choose to morph the field into a TOC, but then this code will just skip doing it. So the next bug will be "I choose morph to a TOC, and nothing happens." I believe the right place for this is in the menu construction. If that's not possible for whatever reason, then we have to pop a message to them telling them that the action doesn't make sense / cannot be done.